mantidproject/mantid
# Enabling and disabling reflectometry GUI components

This PR improves functionality to enable/disable GUI components on the `ISIS Reflectometry` interface and `DataProcessorWidget`.

**To test:**

- Check that the following GUIs still operate correctly:
    - `ISIS Reflectometry`
    - `ISIS SANS v2 experimental`
    - `DataProcessorInterface` (can be enabled by adding `Utilities/DataProcessorInterface.py` to `mantidqt.python_interfaces` in `Framework/Properties/Mantid.properties.template`
- Menu items in the toolbar, context menu and main menus are disabled/enabled according to whether processing is in progress. Try processing some runs and check that:
  - The pause button is disabled while idle and enabled while processing. Check that hitting Pause stops reduction.
  - Buttons that do not change reduction settings are always enabled: plotting, group selection, expand/collapse rows (for all interfaces using the DataProcessorWidget). Check that these still work while processing is in progress
  - For  `ISIS Reflectometry`, the save/export options and slit calculator in the Reflectometry menu should always be enabled
  - Other buttons are enabled while idle and disabled while processing
  - The toolbar buttons have the same enabled/disabled state as their counterparts in the context menu (and also the Edit menu for reflectometry - note that this PR doesn't cover enabling/disabling main menu items in the SANS GUI).
  - Double-clicking on a cell in the DataProcessorWidget normally allows you to edit its contents but this should be disabled during processing.

Fixes #21215 

Note that this also adds most of the functionality for #19951 but the problem from that issue of enabling the pause button too soon is not yet fixed.

**Release Notes** 
*Added to the release notes*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
